File: src/main/java/com/innercircle/project_one/survey/domain/SurveyVersion.java
package com.innercircle.project_one.survey.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class SurveyVersion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long version;

    @OneToOne
    @JoinColumn(name = "survey_id")
    private Survey survey;


    public SurveyVersion(Long version, Survey survey) {
        this.version = version;
        this.survey  = survey;
    }

    public void setVersion(Long version){
        this.version = version;
    }

}

File: src/main/kotlin/innerCircle/survey/entity/SurveyVersion.kt
package innerCircle.survey.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalDateTime

@Entity
data class SurveyVersion(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @ManyToOne @JoinColumn(name = "survey_id")
    val survey: Survey,
    val version: Int,
    val createdAt: LocalDateTime = LocalDateTime.now()
){
    companion object {
        fun create(survey: Survey): SurveyVersion {
            return SurveyVersion(
                survey = survey,
                version = 1
            )
        }
    }
}
